libr/core/cmd.c

```diff
@@ -5953,6 +5983,356 @@ DEFINE_HANDLE_TS_FCN(iter_hit_command) {
+	int fd = r_io_fd_get_current (core->io);
+	// only iterate maps of current fd
+	RList *maps = r_io_map_get_for_fd (core->io, fd);
+	RIOMap *map;
+	bool res = true;
+	if (maps) {
+		RListIter *iter;
+		r_list_foreach (maps, iter, map) {
+			r_core_seek (core, map->itv.addr, 1);
+			r_core_block_size (core, map->itv.size);
+			res &= handle_ts_command (state, command);
+		}
+		r_list_free (maps);
+	}
+	return res;
+}
+
+DEFINE_HANDLE_TS_FCN(foreach_dbgmap_command) {
+	RCore *core = state->core;
+	TSNode command = ts_node_named_child (node, 0);
+	RDebug *dbg = core->dbg;
+	bool res = true;
+	if (dbg && dbg->h && dbg->maps) {
+		RDebugMap *map;
+		RListIter *iter;
+		r_list_foreach (dbg->maps, iter, map) {
+			r_core_seek (core, map->addr, 1);
+			res &= handle_ts_command (state, command);
+		}
+	}
+	return res;
+}
+
+DEFINE_HANDLE_TS_FCN(foreach_flag_command) {
+	RCore *core = state->core;
+	TSNode command = ts_node_named_child (node, 0);
+	TSNode filter_node = ts_node_named_child (node, 1);
+	char *glob = NULL;
+	if (!ts_node_is_null (filter_node)) {
+		glob = ts_node_sub_string (filter_node, state->input);
+	}
+	ut64 off = core->offset;
+	ut64 obs = core->blocksize;
+	RList *flags = r_list_newf (free);
+	r_flag_foreach_glob (core->flags, glob, copy_into_flagitem_list, flags);
+	RListIter *iter;
+	RFlagItem *f;
+	bool res = true;
+	r_list_foreach (flags, iter, f) {
+		r_core_block_size (core, f->size);
+		r_core_seek (core, f->offset, 1);
+		res &= handle_ts_command (state, command);
+	}
+	r_core_seek (core, off, 0);
+	r_core_block_size (core, obs);
+	free (glob);
+	return res;
+}
+
+DEFINE_HANDLE_TS_FCN(foreach_function_command) {
+	RCore *core = state->core;
+	TSNode command = ts_node_named_child (node, 0);
+	TSNode filter_node = ts_node_named_child (node, 1);
+	char *filter = NULL;
+	if (!ts_node_is_null (filter_node)) {
+		filter = ts_node_sub_string (filter_node, state->input);
+	}
+	ut64 obs = core->blocksize;
+	ut64 offorig = core->offset;
+	RAnalFunction *fcn;
+	RList *list = core->anal->fcns;
+	RListIter *iter;
+	bool res = true;
+	r_cons_break_push (NULL, NULL);
+	r_list_foreach (list, iter, fcn) {
+		if (r_cons_is_breaked ()) {
+			break;
+		}
+		if (!filter || r_str_glob (fcn->name, filter)) {
+			r_core_seek (core, fcn->addr, 1);
+			r_core_block_size (core, r_anal_function_linear_size (fcn));
+			res &= handle_ts_command (state, command);
+		}
+	}
+	r_cons_break_pop ();
+	r_core_block_size (core, obs);
+	r_core_seek (core, offorig, 1);
+	return res;
+}
+
+DEFINE_HANDLE_TS_FCN(foreach_thread_command) {
+	RCore *core = state->core;
+	TSNode command = ts_node_named_child (node, 0);
+	RDebug *dbg = core->dbg;
+	bool res = true;
+	if (dbg && dbg->h && dbg->h->threads) {
+		int origpid = dbg->pid;
+		RDebugPid *p;
+		RList *list = dbg->h->threads (dbg, dbg->pid);
+		if (!list) {
+			return false;
+		}
+		RListIter *iter;
+		r_list_foreach (list, iter, p) {
+			r_core_cmdf (core, "dp %d", p->pid);
+			r_cons_printf ("PID %d\n", p->pid);
```

Comment:
unnecessary printf

Reply:
mmm.. I'm just mimicking existing commands and mostly copy-pasted existing code. It was already doing this printf and I copied it.